app/src/main/java/io/gnosis/safe/ui/settings/owner/selection/DerivedOwnerListAdapter.kt

```diff
@@ -65,40 +84,82 @@ class DerivedOwnerListAdapter() : PagingDataAdapter<Solidity.Address, DerivedOwn
 
     override fun getItemViewType(position: Int): Int =
         if (position == 0) {
-            AccountItemViewType.DEFAULT_OWNER.ordinal
+            if (getItem(position)?.disabled == true) {
+                AccountItemViewType.DISABLED_DEFAULT_OWNER.ordinal
+            } else {
+                AccountItemViewType.DEFAULT_OWNER.ordinal
+            }
         } else {
-            AccountItemViewType.OWNER.ordinal
+            if (getItem(position)?.disabled == true) {
+                AccountItemViewType.DISABLED_OWNER.ordinal
+            } else {
+                AccountItemViewType.OWNER.ordinal
+            }
         }
 
-
-    private fun getSelectedOwnerIndex(selectedOwnerPosition: Int): Long = selectedOwnerPosition.toLong()
+    fun getSelectedOwnerIndex(): Long = selectedOwnerPosition.toLong()
 
     interface OnOwnerItemClickedListener {
         fun onOwnerClicked(ownerIndex: Long)
     }
 
     enum class AccountItemViewType {
         DEFAULT_OWNER,
-        OWNER
+        OWNER,
+        DISABLED_OWNER,
+        DISABLED_DEFAULT_OWNER
     }
 
     abstract class BaseOwnerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        abstract fun bind(address: Solidity.Address, position: Int)
+        abstract fun bind(address: OwnerHolder, position: Int)
+    }
+
+    inner class DisabledOwnerViewHolder(private val binding: ItemOwnerSelectionDisabledOwnerBinding) : BaseOwnerViewHolder(binding.root) {
```

Comment:
it seems like a redundancy to me. Can OwnerViewHolder handle both disabled and enabled state?

Reply:
As I said I had problems getting the two different (with and without label) layouts into one. That's why I decided to have two layouts.